logger/logger.go

```diff
@@ -0,0 +1,101 @@
+package logger
+
+import (
+	"os"
+
+	logrus "github.com/tidepool-org/platform/Godeps/_workspace/src/github.com/Sirupsen/logrus"
+	"github.com/tidepool-org/platform/Godeps/_workspace/src/github.com/satori/go.uuid"
+)
+
+//Logger interface
+type Logger interface {
+	GetNamed(name string) Logger
+	WithField(key string, value interface{}) Logger
+	AddTrace(id string)
+	AddTraceUUID()
+	Debug(args ...interface{})
+	Info(args ...interface{})
+	Warn(args ...interface{})
+	Error(args ...interface{})
+	Fatal(args ...interface{})
+}
+
+const traceID = "traceID"
+
+//PlatformLog type
+type PlatformLog struct {
+	fields   map[string]interface{}
+	internal *logrus.Logger
+}
+
+//Log is an initialised PlatformLog instance
+var Log *PlatformLog
+
+//init will create a std logger that implements the `Logger` interface with all methods exported below
+//your other option is to roll your own logger again making sure that it implements the `Logger` interface
+func init() {
+	Log = setup()
+}
+
+func setup() *PlatformLog {
+
+	logger := logrus.New()
+	logger.Out = os.Stdout
+	logger.Formatter = new(logrus.JSONFormatter)
+	logger.Level = logrus.ErrorLevel
+
+	return &PlatformLog{
+		internal: logger,
+		fields:   make(map[string]interface{}),
+	}
+
+}
+
+//GetNamed return a named instance of the logger
+func (log *PlatformLog) GetNamed(name string) Logger {
+	named := setup()
+	named.WithField("name", name)
+	return named
+}
+
+//AddTrace will add a trace id to the logs
+func (log *PlatformLog) AddTrace(id string) {
```

Comment:
This adds a single field for ALL log statements.  The problem is that it uses the global `Log` and so simultaneous requests will conflict with each other.  If you attached it to just a specific instance of a logger that was only used for a single request (and then discarded) then this might work.

Also consider that we will likely have at least two "tracing" mechanisms.  One for each request and one from the client for each "session" of requests.